cmd/client_test.go

```diff
@@ -21,53 +19,58 @@ package cmd
 
 import (
 	"github.com/golang/glog"
-	"os"
 	"testing"
 
 	"github.com/golang/protobuf/proto"
+	"github.com/google/cloudprober"
 	"github.com/google/cloudprober/examples/extensions/myprober/myprobe"
 	probes_configpb "github.com/google/cloudprober/probes/proto"
 	targetspb "github.com/google/cloudprober/targets/proto"
-	"github.com/google/cloudprober"
-	"github.com/google/cloudprober/web"
 
 	"github.com/googleinterns/step224-2020/hermes"
 )
 
 const (
 	cpCfg     string = "grpc_port:9314" // Cloudprober config
 	rpcServer string = "localhost:9314" // Cloudprober gRPC server address
+)
 
-	testProbe0 string = "testExtension0"
-	testProbe1 string = "testExtension1"
+var (
+	testProbes = []string{
+		"testExtension0",
+		"testExtension1"}
 )
 
-//setupHermes initialises an instance of Cloudprober so that RPCs can be sent to it.
-func setupHermes(hermes *hermes.Hermes) {
-	err := hermes.InitialiseCloudproberFromConfig(cpCfg)
+func setupCloudproberAndClient() (*hermes.Hermes, *CloudproberClient) {
+	// Initialise Cloudprober
+	testHermes := &hermes.Hermes{}
+	err := testHermes.InitialiseCloudproberFromConfig(cpCfg)
 	if err != nil {
 		glog.Fatalf("Cloudprober could not be initialised, err: %v", err)
 	}
-}
 
-// TestMain is used to setup Cloudprober before tests are run.
-// This is necessary for there to be a gRPC server to make RPCs to.
-func TestMain(m *testing.M) {
-	test_hermes := &hermes.Hermes{}
+	// Start running Cloudprober instance from Hermes context
+	cloudprober.Start(testHermes.Ctx)
 
-	setupHermes(test_hermes)              // Set up Hermes and initialise Cloudprober
-	defer test_hermes.CancelCloudprober() // Cancel Cloudprober after tests are completed
+	// Initialise Cloudprober gRPC client
+	var client *CloudproberClient
```

Comment:
Just use `:=` and don't predeclare this.

Also, we don't need the comment.

Reply:
Amended, removed.